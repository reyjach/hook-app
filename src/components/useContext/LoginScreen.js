import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {


  const { setUser } = useContext( UserContext );

  const handleClick = () => {
    setUser({
      id: 123,
      name: 'carlos',
      email: 'carlos23@gmail.com'
    })
  }

  return (
    <div>
        <h1> LoginScreen </h1>
        <hr/>

        <button 
          className="btn btn-primary"
          onClick={ handleClick }>
          Login
        </button>
    </div>
  )
}
