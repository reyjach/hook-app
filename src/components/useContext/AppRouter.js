import React from 'react';
import { BrowserRouter as Router, Routes, Route} from "react-router-dom";

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';


export const AppRouter = () => {
  return (
    <Router>
        
        <NavBar></NavBar>
        <div className="container">
            <Routes>
                <Route index path="/about" element={<AboutScreen />}/>

                <Route index path="/login" element={<LoginScreen />}/>

                <Route index path="/" element={<HomeScreen />} />

                <Route path="*" element={<HomeScreen />} />

            </Routes>
        </div>
    </Router>
  )
}
