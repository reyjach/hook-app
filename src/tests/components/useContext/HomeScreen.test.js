import { mount } from "enzyme";
import { HomeScreen } from "../../../components/useContext/HomeScreen";
import { UserContext } from "../../../components/useContext/UserContext";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme';

describe('Pruebas en el componente HomeScreen', () => { 

    Enzyme.configure({ adapter: new Adapter() });

    const user = {
        name:'Fernando',
        email: 'fernando342@gmail.com'
    }

    const wrapper = mount( 

        <UserContext.Provider value={[
            user
        ]}>
            <HomeScreen /> 
        </UserContext.Provider>

    );

    test('Debe de mostrarse correctamtente', () => { 
            
        expect( wrapper ).toMatchSnapshot();

     });

 });