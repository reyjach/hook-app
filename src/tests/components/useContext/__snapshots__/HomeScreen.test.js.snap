// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente HomeScreen Debe de mostrarse correctamtente 1`] = `
<div>
  <h1>
     HomeScreen 
  </h1>
  <hr />
  <pre />
</div>
`;
